src/browser/index.ts

```diff
@@ -6,32 +6,44 @@ import { getNonce, createStyleNode, appendStyleNode } from './dom';
 import { createMultiThemeCustomizer } from '../shared/declaration/customizer';
 import { getContexts, getThemeFromPreset } from '../shared/theme/validate';
 
-export interface ApplyThemeParams {
+export interface GenerateThemeStylesheetParams {
   override: Override;
   preset: ThemePreset;
   baseThemeId?: string;
   targetDocument?: Document;
 }
 
-export interface ApplyThemeResult {
-  reset: () => void;
-}
-
-export function applyTheme(params: ApplyThemeParams): ApplyThemeResult {
-  const { override, preset, baseThemeId, targetDocument } = params;
+export function generateThemeStylesheet(params: GenerateThemeStylesheetParams): string {
```

Comment:
My intention was to expose `generateThemeStylesheet` from the `theming-runtime` package. Looks like if I move it to `src/build` it would be instead exposed via `theming-build`. IMO that doesn't fit with the use case for this function -- but are you saying regardless of the package names the packages are really `theming-browser` and `theming-node`? FWIW, in my testing simply importing `generateThemeStylesheet` on the server doesn't break anything because the DOM API dependencies would only become an issue if `applyTheme` were actually called server side. (But perhaps you're concerned with future potential of needing to use side effects in this module?) Also, while it makes the most sense to call `generateThemeStylesheet` server side, there's technically nothing stopping anyone from calling it client side

Either way, happy to move it once you confirm

Reply:
I see, so you are going to call this function in isomorphic code (something like `app/layout.js` in next). You do not want to pull build time dependencies in that closure, that's true.

I guess, we will end up exposing it in both, but let's keep your current option with `src/browser/index.ts`

Reply:
P.S. While we are at this thread, can we also have more tests on `generateThemeStylesheet` standalone?